File: 0370-range-addition/0370-range-addition.java
class Solution {
    public int[] getModifiedArray(int length, int[][] updates) {
        int[] diff = new int[length + 1];
        for(int[] x : updates) {
            diff[x[0]] += x[2];
            diff[x[1] + 1] -= x[2];
        }
        int val = 0;
        int[] res = new int[length];
        
        for(int i = 0; i < length; ++i) {
            val += diff[i];
            res[i] = val;
        }
        
        return res;
    }
}

File: 0370-range-addition/0370-range-addition.py
class Solution:
    def getModifiedArray(self, length: int, updates: List[List[int]]) -> List[int]:
        diff = [0] * (length + 1);
        
        for u in updates:
            diff[u[0]] += u[2]
            diff[u[1] + 1] -= u[2]
            
        res = [0] * length
        val = 0
        
        for i in range(length):
            val += diff[i];
            res[i] = val
        return res;